typing/patterns.mli

```diff
@@ -37,34 +37,57 @@ module Non_empty_row : sig
 end
 
 module Simple : sig
-  type view = [
+
+  type core_view = [
     | `Any
     | `Constant of constant
     | `Tuple of pattern list
     | `Construct of
         Longident.t loc * constructor_description * pattern list
     | `Variant of label * pattern option * row_desc ref
-    | `Record of
-        (Longident.t loc * label_description * pattern) list * closed_flag
     | `Array of pattern list
     | `Lazy of pattern
   ]
+
+  type record_view = [
+      `Record of
+        (Longident.t loc * label_description * pattern) list * closed_flag
+  ]
+
+  type expanded_record_view = [
+      `Expanded_record of
+        (Longident.t loc * label_description * pattern) list
+  ]
+
+  type view = [ core_view | expanded_record_view ]
+  type nonexpanded_view = [ view | record_view ]
   type pattern = view pattern_data
+  type nonexpanded_pattern = nonexpanded_view pattern_data
+
+  val omega : [> `Any ] pattern_data
 
-  val omega : [> view ] pattern_data
 end
 
 module Half_simple : sig
   type view = [
     | Simple.view
     | `Or of pattern * pattern * row_desc option
   ]
+
+  type nonexpanded_view = [
+    | Simple.nonexpanded_view
+    | `Or of pattern * pattern * row_desc option
+  ]
```

Comment:
Could the `` `Or`` constructor have its own type declaration to not repeat it twice in `view` and `expanded_view`?